direct/tpm2/policy_test.go

```diff
@@ -281,6 +282,139 @@ func TestPolicyOrUpdate(t *testing.T) {
 	}
 }
 
+func getExpectedPCRDigest(t *testing.T, thetpm transport.TPM, selection tpml.PCRSelection, hashAlg tpm.AlgID) []byte {
+	t.Helper()
+	pcrRead := PCRRead{
+		PCRSelectionIn: selection,
+	}
+
+	pcrReadRsp, err := pcrRead.Execute(thetpm)
+	if err != nil {
+		t.Fatalf("Failed to read PCRSelection")
+	}
+
+	var expectedVal []byte
+	for _, digest := range pcrReadRsp.PCRValues.Digests {
+		expectedVal = append(expectedVal, digest.Buffer...)
+	}
+
+	cryptoHashAlg, err := hashAlg.Hash()
+	hash := cryptoHashAlg.New()
+	hash.Write(expectedVal)
+	return hash.Sum(nil)
+}
+
+func TestPolicyPCR(t *testing.T) {
+	thetpm, err := simulator.OpenSimulator()
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+	defer thetpm.Close()
+
+	PCRs, err := CreatePCRSelection([]int{0, 1, 2, 3, 7})
+	if err != nil {
+		t.Fatalf("Failed to create PCRSelection")
+	}
+	selection := tpml.PCRSelection{
+		PCRSelections: []tpms.PCRSelection{
+			{
+				Hash:      tpm.AlgSHA1,
+				PCRSelect: PCRs,
+			},
+		},
+	}
+
+	expectedDigest := getExpectedPCRDigest(t, thetpm, selection, tpm.AlgSHA1)
+	t.Logf("expectedDigest=%x", expectedDigest)
+
+	wrongDigest := sha1.Sum(expectedDigest[:])
+
+	tests := []struct {
+		name              string
+		authOption        []AuthOption
+		pcrDigest         []byte
+		callShouldSucceed bool
+	}{
+		{"TrialCorrect", []AuthOption{Trial()}, expectedDigest, true},
+		{"TrialIncorrect", []AuthOption{Trial()}, wrongDigest[:], true},
+		{"TrialEmpty", []AuthOption{Trial()}, nil, true},
+		{"RealCorrect", nil, expectedDigest, true},
+		{"RealIncorrect", nil, wrongDigest[:], false},
+		{"RealEmpty", nil, nil, true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sess, cleanup2, err := PolicySession(thetpm, tpm.AlgSHA1, 16, tt.authOption...)
+
+			if err != nil {
+				t.Fatalf("setting up policy session: %v", err)
+			}
+
+			policyPCR := PolicyPCR{
+				PolicySession: sess.Handle(),
+				PcrDigest: tpm2b.Digest{
+					Buffer: tt.pcrDigest,
+				},
+				Pcrs: selection,
+			}
+
+			err = policyPCR.Execute(thetpm)
+			if tt.callShouldSucceed {
+				if err != nil {
+					t.Fatalf("executing PolicyPCR: %v", err)
+				}
+			} else {
+				if err == nil {
+					t.Fatalf("failing test PolicyPCR is passing")
+				}
+				return
+			}
+
+			pgd := PolicyGetDigest{
+				PolicySession: sess.Handle(),
+			}
+			want, err := pgd.Execute(thetpm)
+			if err != nil {
+				t.Fatalf("executing PolicyGetDigest: %v", err)
+			}
+
+			// If the pcrDigest is empty: see TPM 2.0 Part 3, 23.7.
+			if tt.pcrDigest == nil {
+				expectedDigest := getExpectedPCRDigest(t, thetpm, selection, tpm.AlgSHA1)
+				t.Logf("expectedDigest=%x", expectedDigest)
+
+				// Create a populated policyPCR for the PolicyCalculator
+				policyPCR = PolicyPCR{
```

Comment:
nit: you can simplify by just setting `policyPCR.PcrDigest.Buffer = expectedDigest[:]` instead of recreating the entire object